### PR TITLE
protoc-gen-go-grpc: adopt

### PR DESCRIPTION
**Pull request by aaronjheng**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/pr/protoc-gen-go-grpc/package.nix [hunk 1]
```diff
@@ -6,22 +6,28 @@
 
 buildGoModule rec {
   pname = "protoc-gen-go-grpc";
-  version = "1.3.0";
+  version = "1.5.1";
```
Comment (thiagokokada) [inline pkgs/by-name/pr/protoc-gen-go-grpc/package.nix:9]: Please separate this in different commits, one for the adoption and another for the bump.

**Comment by aaronjheng** [comment]
`opensnitch` will use a variant with an older version for now.

**Comment by aaronjheng** [comment]
> It would be good to raise an issue upstream, too.

I don't see this as an issue. Usually, generated code is committed to the repository as part of the version. However, some projects choose to generate code at build time. In such cases, there will inevitably be situations where the generated code is incompatible with the dependencies, due to differences in the versions of the generation tools.

> Also, did you confirm that it really needs 1.3.0 and can't be bumped to e.g. 1.4.x etc?

I can check it, but I don't think it's particularly important. As I mentioned earlier, the compatibility between code generation tools and libraries is quite complex, especially with gRPC and Protobuf. Given the current situation, I believe it's enough to stay aligned with the upstream.

**Comment by SuperSandro2000** [comment]
@ofborg build opensnitch protoc-gen-go-grpc
